OSSM_13620_nftables (#111267)
openshift/openshift-docs: modules/ossm-nftables-migration-ambient.adoc

== Enabling `nftables` support
To enable `nftables` support, you must set the `spec.values.global.nativeNftables` field to `true` in both the `{istio}` and `IstioCNI` resources. This configuration is compatible with earlier versions of {op-system-base-full} and allows for mixed-version clusters during migration.

== How {SMProductShortName} initializes with `nftables` enabled
The {istio} CNI agent checks for existing `iptables` artifacts on the host during startup. The result of this check determines the networking backend configuration based on environment:

* **Clean environments:** On nodes with no existing `iptables` artifacts, such as new {op-system-base-full} 10 hosts, the CNI agent automatically uses the `nftables` backend.
* **Existing environments:** On nodes where `iptables` was active, the CNI agent might detect existing artifacts. To prevent networking conflicts, the agent overrides the `nftables` setting and continues to use the `iptables` backend.

== Final migration step for existing nodes

> To complete a migration from `iptables` to `nftables` on existing nodes, you must clear existing networking artifacts. Rebooting the cluster nodes is the most effective method to ensure a clean networking state for the `nftables` backend. 

Reviewer: There is no easy way for the users to clear the existing networking artifacts on the nodes, so lets rephrase the sentence as follows.

```suggestion
To migrate existing nodes from `iptables` to `nftables`, the nodes should be rebooted to ensure a clean networking state.
```

Author: Done!
